src/cdomains/floatDomain.ml

```diff
@@ -37,130 +38,210 @@ module type FloatDomainBase = sig
-
-  let gt = eval_int_binop gt
-
-  let le _ _ = failwith "todo - le"
-
-  let ge _ _ = failwith "todo - ge"
-
-  let eq = eval_int_binop eq
-
-  let ne = eval_int_binop ne
+  let add op1 op2 = 
+    match op1, op2 with
+    | Some (l1, h1), Some (l2, h2) -> Some (add Down l1 l2, add Up h1 h2)
+    | _ -> None
+
+  let sub op1 op2 = 
+    match op1, op2 with
+    | Some (l1, h1), Some (l2, h2) -> norm @@ Some (sub Down l1 l2, sub Up h1 h2)
+    | _ -> None
+
+  let mul op1 op2 = 
+    match op1, op2 with
+    | Some (l1, h1), Some (l2, h2) -> 
+      let mul1u = mul Up l1 l2 in
+      let mul2u = mul Up l1 h2 in
+      let mul3u = mul Up h1 l2 in
+      let mul4u = mul Up h1 h2 in
+      let mul1d = mul Down l1 l2 in
+      let mul2d = mul Down l1 h2 in
+      let mul3d = mul Down h1 l2 in
+      let mul4d = mul Down h1 h2 in
+      let high = max (max (max mul1u mul2u) mul3u) mul4u in
+      let low = min (min (min mul1d mul2d) mul3d) mul4d in
+      norm @@ Some (low, high)
+    | _ -> None
+
+  let div op1 op2 = 
+    match op1, op2 with
+    | Some (l1, h1), Some (l2, h2) ->
+      if l2 <= 0. && h2 >= 0. then None
+      else
+        let div1u = div Up l1 l2 in
+        let div2u = div Up l1 h2 in
+        let div3u = div Up h1 l2 in
+        let div4u = div Up h1 h2 in
+        let div1d = div Down l1 l2 in
+        let div2d = div Down l1 h2 in
+        let div3d = div Down h1 l2 in
+        let div4d = div Down h1 h2 in
+        let high = max (max (max div1u div2u) div3u) div4u in
+        let low = min (min (min div1d div2d) div3d) div4d in
+        norm @@ Some (low, high)
+    | _ -> None
+
+  let lt op1 op2 =
+    let (a, b) =
+      match op1, op2 with
+      | Some (l1, h1), Some (l2, h2) -> 
+        if h1 < l2 then (1, 1)
+        else if l1 >= h2 then (0, 0) 
+        else (0, 1)
+      | _ -> (0, 1)
+    in
+    IntDomain.IntDomTuple.of_interval IBool (Big_int_Z.big_int_of_int a, Big_int_Z.big_int_of_int b)
+
+  let gt op1 op2 =
+    let (a, b) =
+      match op1, op2 with
+      | Some (l1, h1), Some (l2, h2) -> 
+        if l1 > h2 then (1, 1)
+        else if h1 <= l2 then (0, 0) 
+        else (0, 1)
+      | _ -> (0, 1)
+    in
+    IntDomain.IntDomTuple.of_interval IBool (Big_int_Z.big_int_of_int a, Big_int_Z.big_int_of_int b)
+
+  let le op1 op2 =
+    let (a, b) =
+      match op1, op2 with
+      | Some (l1, h1), Some (l2, h2) -> 
+        if h1 <= l2 then (1, 1)
+        else if l1 > h2 then (0, 0) 
+        else (0, 1)
+      | _ -> (0, 1)
+    in
+    IntDomain.IntDomTuple.of_interval IBool (Big_int_Z.big_int_of_int a, Big_int_Z.big_int_of_int b)
+
+  let ge op1 op2 =
+    let (a, b) =
+      match op1, op2 with
+      | Some (l1, h1), Some (l2, h2) -> 
+        if l1 >= h2 then (1, 1)
+        else if h1 < l2 then (0, 0) 
+        else (0, 1)
+      | _ -> (0, 1)
+    in
+    IntDomain.IntDomTuple.of_interval IBool (Big_int_Z.big_int_of_int a, Big_int_Z.big_int_of_int b)
+
+  let eq op1 op2 = 
```

Comment:
`eq` could probably also be implemented as `not neq` or vice versa. I see that this version is more expressive, however i would prefer less code to maintain and use a few negations where possible. But this is open to you, maybe also better to postpone this until we have all edge-cases covered with test, in order to quickly find out, whether "simplifications" don't break anything.